common/beerocks/bcl/source/network/network_utils.cpp

```diff
@@ -363,6 +365,95 @@ static int parseRoutes(struct nlmsghdr *nlHdr, std::shared_ptr<route_info> rtInf
     return (0);
 }
 
+std::shared_ptr<std::unordered_map<std::string, std::string>>
+network_utils::get_arp_table(bool mac_as_key)
+{
+    auto arp_table_ptr = std::make_shared<std::unordered_map<std::string, std::string>>();
+    auto arp_table     = *arp_table_ptr.get();
+
+    // Allocate a new netlink socket
+    auto nl_socket = nl_socket_alloc();
```

Comment:
It is best to declare a unique_ptr with custom `deleter()`, then later you don't have to worry about freeing the memory manually -
```suggestion
    std::unique_ptr<std::remove_pointer<decltype(nl_socket_alloc())>::type, std::function<void(int*)>> 
        int_ptr(nl_socket_alloc(), nl_socket_free);
```
take a look here:
https://godbolt.org/z/DYBhRf

Reply:
I prefer keeping it as is. It looks better and more readable.
There are some things you can do on CPP that can give you the creeps just looking at them.
This is one of them :)

Reply:
`std::remove_pointer<decltype(nl_socket_alloc())>::type` is used just because I was too lazy to look for the return type of `nl_socket_alloc()` - if you know the type, just write it.
`std::function<void(int*)>` is required as this is the interface of `nl_socket_free` and decltype of it doesn't work (I don't know why yet).
Though it looks frightening this is much safer and better code. It removes 90% of the errors happening when working with pointers.
I would really like us all to start writing safer and better code.